lib/provider/search_screen_provider.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:e_commerce_application/repository/product_repository.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:e_commerce_application/models/product.dart';
+import 'package:get_it/get_it.dart';
+
+class SearchScreenProvider extends ChangeNotifier {
+  final productRepository = GetIt.instance<ProductRepository>();
+  List<Product> productList = [];
+  bool noResult = false;
+
```

Comment:
Remove additional space. 1 line of space is enough between variables and function